src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -69,6 +81,7 @@ public BinaryJedisCluster(Set<HostAndPort> jedisClusterNode, int connectionTimeo
     this.connectionHandler = new JedisSlotBasedConnectionHandler(jedisClusterNode, poolConfig,
         connectionTimeout, soTimeout, user, password, clientName);
     this.maxAttempts = maxAttempts;
+    this.maxTotalRetriesDuration = Duration.ofMillis(soTimeout);
```

Comment:
Why are we connecting soTimeout with maxTotalRetriesDuration?
I think we should have a separate argument for the maxTotalRetriesDuration and perhaps for enabling backoff at all to avoid backward issues (at least in 3.6)

Reply:
@gkorland 

> Why are we connecting soTimeout with maxTotalRetriesDuration?

Because max duration for one single try IS EQUALS TO soTimeout.